### PR TITLE
MB 12762 Add violations section to review modal and 'view report'

### PR DESCRIPTION
**Pull request by mandaem**
## Summary

This PR will allow user to see violation details in both the preview confirmation modal as well as in the 'view report' generated after a report is successfully submitted.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Login as a QAE and go to an evaluation report (new or edit). 
2. If no violations: if none have been selected, once you've completed and saved the main eval report, you'll click 'review and submit' which will launch modal showing no violations were observed. 
3. If violations selected: once selected and form was saved, as well as serious incident portion completed, you'll navigate back to main eval report for those violations (Until a separate PR is merged which allows 'review and submit' directly from violations page). 
4. Then after you click 'no' radio btn for violations, you'll click on 'review and submit' . You should now see the violations you selected as well as the information for the serious incident portion. 
5. If you don't see the violations after you selected them for that eval report, ensure you've saved drafts and are in the correct eval report, which allows violations data to be pulled into modal.
 

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [x] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

## Screenshots
_Fotor

### CODE REVIEW

**Comment by AnnaGingle** [comment]
Looks good! Approved

**Review by AnnaGingle — APPROVED** [review]
Looks great!

File: migrations/app/schema/20220919160852_create_report_violations_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+		REFERENCES "evaluation_reports",
+	violation_id uuid NOT NULL
+		CONSTRAINT report_violations_violation_id_fkey
+		REFERENCES "pws_violations"
+);
+
+CREATE INDEX report_violations_report_id_idx ON report_violations (report_id);
+
+COMMENT ON TABLE report_violations IS 'Associates PWS Violations with QAE evaluation report.';
+COMMENT ON COLUMN report_violations.report_id IS 'Report ID of the report that violations will be assiocated with.';
```
Comment (mboren) [inline migrations/app/schema/20220919160852_create_report_violations_table.up.sql:14]: little typo in both the column comments: `assiocated` should be `associated`

**Comment by Ryan-Koch** [comment]
If this gets a merge from master it should remove some diffs that were pull overs from the other PR that merged.

File: migrations/app/schema/20220919160852_create_report_violations_table.up.sql (hunk 1 above)
Comment (mboren) [reply migrations/app/schema/20220919160852_create_report_violations_table.up.sql:14]: oops, this change is not really from this PR

File: src/components/Office/EvaluationViolationsForm/SelectedViolation/SelectedViolation.jsx [hunk 2]
```diff
@@ -16,9 +16,11 @@ const SelectedViolation = ({ violation, unselectViolation }) => {
...
         </p>
       </div>
-      <Button type="button" unstyled onClick={() => unselectViolation(violation.id)} role="button">
-        Remove
-      </Button>
+      {!isReadOnly && (
+        <Button type="button" unstyled onClick={() => unselectViolation(violation.id)} role="button">
+          Remove
+        </Button>
+      )}
```
Comment (PDickeyTruss) [inline src/components/Office/EvaluationViolationsForm/SelectedViolation/SelectedViolation.jsx:23]: Suggestion: Probably should add a test for this new prop/functionality

File: src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx [hunk 3]
```diff
@@ -62,11 +80,11 @@ const EvaluationReportPreview = ({ evaluationReport, mtoShipments, moveCode, cus
   };
 
   return (
-    <div className={bordered ? styles.bordered : styles.borderless} data-testid="EvaluationReportPreview">
+    <div className={styles.evaluationReportPreview} data-testid="EvaluationReportPreview">
       <div>
         <div className={styles.titleSection}>
           <div className={styles.pageHeader}>
-            <h1>{`${isShipment ? 'Shipment' : 'Counseling'} report`}</h1>
+            <h1>{reportType === 'SHIPMENT' ? 'Shipment' : 'Move'} report</h1>
```
Comment (PDickeyTruss) [inline src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx:87]: Looks like a bit of a regression here, probably want to switch back to using the constant rather than a string and likely the variable (`isShipment`) as well since we use this a few times. 

```
import EVALUATION_REPORT_TYPE from 'constants/evaluationReports';
const isShipment = evaluationReport.type === EVALUATION_REPORT_TYPE.SHIPMENT;
...
<h1>{isShipment ? 'Shipment' : 'Move'} report</h1>

```

File: src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx [hunk 4]
```diff
@@ -3,23 +3,41 @@ import * as PropTypes from 'prop-types';
...
+  const showIncidentDescription = evaluationReport.SeriousIncident;
+  let seriousIncident = '';
+  const hasViolations = reportViolations && reportViolations.length > 0;
+  const incidentDescription = evaluationReport.SeriousIncidentDesc;
+
+  if (showIncidentDescription) {
+    seriousIncident = 'yes';
+  } else {
+    seriousIncident = 'no';
+  }
```
Comment (PDickeyTruss) [inline src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx:40]: Suggestion: This is pretty hard to follow (excluding line 33), maybe some comments or renaming would help out?  I think the outcome of this logic here is: 1) determine if we should show the serious incident description and 2) Map true/false to yes/no for the serious incident. I think that means these lines coujld be as simple as:
```
  const isShipment = evaluationReport?.type === EVALUATION_REPORT_TYPE.SHIPMENT;
  const hasViolations = reportViolations && reportViolations.length > 0;
  const showIncidentDescription = evaluationReport?.seriousIncident;

  let mtoShipmentsToShow;
  if (evaluationReport.shipmentID) {
    ...
```

Then we can use that booleans and just display what we need down lower, something like:
```
 {hasViolations && (
              <>
                <div className={classnames(descriptionListStyles.row)}>
                  <dt className={styles.violationsLabel}>Serious incident</dt>
                  <dd className={styles.violationsRemarks}>{evaluationReport?.seriousIncident ? 'yes' : 'no'}</dd>
                </div>
                {showIncidentDescription && (
                  <div className={classnames(descriptionListStyles.row)}>
                    <dt className={styles.violationsLabel}>Serious incident description</dt>
                    <dd className={styles.violationsRemarks}>{evaluationReport?.seriousIncidentDesc}</dd>
                  </div>
                )}
              </>
            )}
```

File: src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx (hunk 3 above)
Comment (mandaem) [reply src/components/Office/EvaluationReportPreview/EvaluationReportPreview.jsx:87]: great catch, thanks!

File: src/components/Office/DefinitionLists/EvaluationReportList.jsx [hunk 5]
```diff
@@ -36,13 +36,13 @@ const EvaluationReportList = ({ evaluationReport }) => {
             {evaluationReport.locationDescription || ''}
           </dd>
         </div>
-        {evaluationReport.travelTimeMinutes && (
+        {evaluationReport.travelTimeMinutes >= 0 && (
```
Comment (Ryan-Koch) [inline src/components/Office/DefinitionLists/EvaluationReportList.jsx:39]: Thanks for including this fix!
